### ISSUE TITLE
Stats not showing on bike+

### ISSUE THREAD

**Issue by corporate-gadfly** [selalipop/grupetto#3]
TY for the software.

I have a bike+ with Peloton App 2.6.1577. Grupetto shows up fine as the overlay, but all the stats show hyphens only.

I have a Java programming background _(don't know kotlin much)_ and I'm able to use `adb` as well. Let me know if I can assist in a diagnosis.

**Comment by corporate-gadfly** [comment]
@selalipop Hope the week is going well for you. Any update on this?

**Comment by parthtapcart** [comment]
@corporate-gadfly I just installed on my Bike+ yesterday and am having the same issue :( I tried the bike restart as @selalipop mentions in the main ReadMe, but it didn't change anything...

**Comment by selalipop** [comment]
Hi, sorry I missed these messages! Could you try posting your ADB Logcat output when starting the app @corporate-gadfly 

That'll tell me if the service Grupetto is using still exists on the Bike+ and some format changed, or if a new method of reaching the sensor is needed

**Comment by parthtapcart** [comment]
Hey @selalipop, here's an attachment of my logs upon starting the app, (I have a Bike+ as well)
[logcat.txt](https://github.com/selalipop/grupetto/files/11279106/logcat.txt)

**Comment by sandra-nguyen** [comment]
omg I have the same issue! I never comment on stuff but felt like grupetto is the saving grace i'd need to have stats show while on netflix etc. I also have a bike+ so wonder if that's the root cause difference. Would love to hear if there's a solution to this! :)

**Comment by selalipop** [comment]
Hi, sorry for the late reply but based on the logs the service doesn't exist on the Bike+

This isn't something I'd be able to fix right now, but if anyone's interested enough to do some digging, there might be a service that's still exposing the sensor data

Finding out would take dumping the Peloton APKs from the tablet

**Comment by ChrisJefferson** [comment]
Hi, I previously pulled the Peloton APKs from my tablet on my bike+, in the hope of doing something like this, but quickly got completely lost and made no progress.

I would be happy to send them over to you -- unfortunately I no long have the bike+, so I couldn't verify anything on the actual bike.

**Comment by jakeholland** [comment]
Hey, I am starting to look into this for my Bike+. I have dumped the com.onepeloton.sensor.diagnostics APK and found that the new service is called "com.onepeloton.affernetservice". I decamped the diagnostics app APK (as it shows sensor data and I'm assuming it's simpler than other peloton apps accessing the data. 

Just getting into how the data is being pulled right now, but looks promising. There are mentions to a V1 API as well in the code as well. The service data models also mention the tread and row too.

**Comment by mgalpy** [comment]
@selalipop @jakeholland I was really looking forward to using this until I read about the Bike+ issue :) Any chance of you adding the sensor data access for the Bike+?

**Comment by joeduvall4** [comment]
It looks like @abhiin1947 has figured out Bike+ integration – I compiled his fork and it works perfectly! Made my night!
https://github.com/abhiin1947/grupetto

**Comment by abhiin1947** [comment]
@joeduvall4 Glad it worked for you as well :) I'll clean up my code and PR it here if I find time this weekend

**Comment by sandra-nguyen** [comment]
this is the best news of my life TYOn Jul 18, 2024, at 10:18 PM, Abhinandan Ramaprasath ***@***.***> wrote:﻿
@joeduvall4 Glad it worked for you as well :) I'll clean up my code and PR it here if I find time this weekend

**Comment by abhiin1947** [comment]
[PR](https://github.com/selalipop/grupetto/pull/5) here for the fix. @selalipop are you able to take a look?

**Comment by marshyrob** [comment]
> It looks like [@abhiin1947](https://github.com/abhiin1947) has figured out Bike+ integration – I compiled his fork and it works perfectly! Made my night! https://github.com/abhiin1947/grupetto

@joeduvall4 can you share the .apk you created? i would like to try this but i have no programming skills.  Would be appreciated very much

**Comment by abhiin1947** [comment]
I can't release it here but here it is on my fork -> https://github.com/abhiin1947/grupetto/releases/tag/v0.2.0

**Comment by marshyrob** [comment]
Amazing!!! thanks so much i truly appreciate it!  Ill give it a shot tomorrow.
________________________________

**Comment by marshyrob** [comment]
> Amazing!!! thanks so much i truly appreciate it!  Ill give it a shot tomorrow.
> […](#)

@abhiin1947- thanks so much for this.  I tried the release but it doesnt work im afraid, it loads up and shows all the metrics as a dash.  When i start to pedal it says sensors may have gone to sleep and to restart.  Ive powered off the bike and started it again but the same issue no data coming through. I started the just ride process and i see the metrics updating on the peloton dash but not in grupetto app.  My tablet is model **PLTN-TRR01-2** (2nd Gen tablet of bike+) not sure if that makes a difference.  Of the 2 files that are in the releases only the one titled **app-debug.apk** installs, the other one fails to install, not sure the difference between the 2?  Can i help in anyway buy providing you a log or something?
